quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java

```diff
@@ -62,7 +64,25 @@ public void shouldCreateApplicationOnJvm() {
         // Start using DEV mode
         app.start();
         app.given().get().then().statusCode(HttpStatus.SC_OK);
-        assertExpectedJavaVersion(getPomFileFromMavenApplication(app));
+        assertExpectedJavaVersion(getPomFileFromMavenApplication(app), getSystemJavaVersion());
+    }
+
+    @Tag("QUARKUS-1472")
+    @Test
+    public void shouldCreateAnApplicationForcingJavaVersion() {
+
+        // Create application
+        QuarkusCliClient.CreateApplicationRequest args = defaults().withExtraArgs("--java=" + truncateCurrentJavaVersion());
+        QuarkusCliRestService app = cliClient.createApplication("app", args);
+
+        // Should build on Jvm
+        QuarkusCliClient.Result result = app.buildOnJvm();
```

Comment:
I think this won't work unless we make sure that `maven-compiler-plugin` uses the same JDK as the one provided by the `--java` param.
Is that even possible? We would need to make sure the JDK is installed, and also specify `JAVA_HOME`.